types/react-native/index.d.ts

```diff
@@ -8467,21 +8467,25 @@ export namespace Animated {
      */
     export function event<T>(argMapping: Array<Mapping | null>, config?: EventConfig<T>): (...args: any[]) => void;
 
+    export interface AnimatedComponent<T extends React.FC | typeof React.Component> extends React.FC<T['props']> {
+      getNode: () => T;
+    }
+
     /**
      * Make any React component Animatable.  Used to create `Animated.View`, etc.
      */
-    export function createAnimatedComponent(component: any): any;
+    export function createAnimatedComponent<T extends React.FC | typeof React.Component>(component: T): AnimatedComponent<T>;
```

Comment:
I wonder if this works too?

```suggestion
    export function createAnimatedComponent<T extends React.ComponentType>(component: T): AnimatedComponent<T>;
```